controllers/harbor/harbor.go

```diff
@@ -30,18 +36,79 @@ func (harbor *HarborReconciler) Reconcile() (*lcm.CRStatus, error) {
 		if errors.IsNotFound(err) {
 			return harbor.Provision()
 		} else {
-			// TODO given clear reason and message.
-			return harborCRNotReadyStatus("", ""), err
+			return harborClusterCRUnknownStatus(), err
 		}
 	}
-	return nil, nil
+	harbor.CurrentHarborCR = &harborCR
+	event := harbor.checkReconcileEvent(harbor.HarborCluster, &harborCR)
+	switch event {
+	case ScalingEvent:
+		return harbor.Scale()
+	case UpdatingEvent:
+		return harbor.Update(harbor.HarborCluster)
+	}
+
+	err = harbor.Get(harbor.getHarborCRNamespacedName(), &harborCR)
+	if err != nil {
+		return harborClusterCRUnknownStatus(), err
+	}
+	return harborClusterCRStatus(&harborCR), nil
 }
 
-func (harbor *HarborReconciler) Delete() (*lcm.CRStatus, error) {
-	panic("implement me")
+func unsetReplicas(harbor *v1alpha1.Harbor) {
```

Comment:
can we build a component array (the elements of the array should implement the same interface) to do the replica setting logic? Code does not look like 'clean', what do you think?